src/internal/animate.ts

```
@@ -21,7 +21,7 @@ export function animateTo(el: HTMLElement, keyframes: Keyframe[], options?: Keyf
 // Parses a CSS duration and returns the number of milliseconds.
 //
 export function parseDuration(delay: number | string) {
-  delay = (delay + '').toLowerCase();
+  delay = delay.toString().toLowerCase();
```

Isn't is safer to use `String(delay)` in case the user passes an invalid option at runtime? I guess we shouldn't guard against every possible user error but the resulting console message will probably seem cryptic. What are your thoughts?

---

At least in this case, I think the only place it is used is used is in the animate component which gets the argument from a CSS custom property. So it should always be a string.

As far as safer at runtime, is the goal of the library to "just work" if the user passes some bad data to the component or should the component "fail fast" to let the user know that they are using it possibly incorrectly?

---

Fair enough.

Generally speaking, we want things to "just work" when possible so things like number <=> string conversions should never throw, for example. But if they pass an array or object into a property that should be a string, I'm fine with that throwing. This is fine as-is.